pkg/stats/stats_recorder.go

```diff
@@ -261,38 +263,13 @@ func (r *recorder) recordIncomingRTCP(latestStats internalStats, v *incomingRTCP
 }
 
 func (r *recorder) Start() {
-	latestStats := &internalStats{}
-	for {
-		select {
-		case <-r.done:
-			return
-		case v := <-r.incomingRTPChan:
-			s := r.recordIncomingRTP(*latestStats, v)
-			latestStats = &s
-
-		case v := <-r.outgoingRTCPChan:
-			s := r.recordOutgoingRTCP(*latestStats, v)
-			latestStats = &s
-
-		case v := <-r.outgoingRTPChan:
-			s := r.recordOutgoingRTP(*latestStats, v)
-			latestStats = &s
-
-		case v := <-r.incomingRTCPChan:
-			s := r.recordIncomingRTCP(*latestStats, v)
-			latestStats = &s
-
-		case r.getStatsChan <- Stats{
-			InboundRTPStreamStats:        latestStats.InboundRTPStreamStats,
-			OutboundRTPStreamStats:       latestStats.OutboundRTPStreamStats,
-			RemoteInboundRTPStreamStats:  latestStats.RemoteInboundRTPStreamStats,
-			RemoteOutboundRTPStreamStats: latestStats.RemoteOutboundRTPStreamStats,
-		}:
-		}
-	}
+	atomic.StoreUint32(&r.running, 1)
```

Comment:
What's the purpose of doing this / why do we need to keep state here? Is there a case in which we attach the interceptor and before `Start` is called, the hooks on the interceptors gets called?

Reply:
I'm not really sure there's a case but as there was both `Start` and `Stop` methods, I figured it was important to behave based on the `running` status.

An alternative would indeed be to remove both `Start` and `Stop` methods and update stats whenever packets are coming/going.